### ISSUE TITLE
Tesseract is sensitive to initial conditions

### ISSUE THREAD

**Issue by Euler-Maskerony** [tesseract-ocr/tesseract#3496]
### Environment

* **Tesseract Version**: both  4 and 5
* **Platform**: Linux arch 5.10.47

### Current Behavior: 
There are two pictures:
1.
![550](https://user-images.githubusercontent.com/62300405/125450007-553a549f-2834-4d80-8270-33d2853c214f.jpg)
2.
![050](https://user-images.githubusercontent.com/62300405/125450055-8244105e-45e4-4ae7-b31d-334a8a40eaa0.jpg)
For the first one result as expected is 550,000 with some noise. The second is quite similar, but the output is 050,000 with noise.

I'm wondering why is it so and what can be done here?

**Comment by nocun** [comment]
In this case I am unable to reproduce the issue, using `fast` dataset I get `550,000` and for `best`: `550000` (language is `eng` and page seg is `tesseract::PSM_SINGLE_LINE`).

Do you get these results in individual runs or when processing in a batch?

**Comment by stweil** [comment]
I get `550,000` plus noise for both images when I add the command line option `--dpi 150`.

**Comment by Euler-Maskerony** [comment]
> Do you get these results in individual runs...?
@nocun 
Yes, but I used psm_auto option because I have a bunch of such images and in general I can't treat them as a single line.

**Comment by wollmers** [comment]
> @nocun Yes, but I used psm_auto option because I have a bunch of such images and in general I can't treat them as a single line.

If you don't know the best fitting `psm` then in my experience `psm 4` is the best option.

Without `psm`:

```
$ tesseract 2.jpg 2  --tessdata-dir /usr/local/share/tessdata txt; cat 2.txt
Tesseract Open Source OCR Engine v5.0.0-alpha-773-gd33ed with Leptonica
[...]
(ou
900,000 & =

```

With `psm 4`:

```
$ tesseract 2.jpg 2  --psm 4 --tessdata-dir /usr/local/share/tessdata txt; cat 2.txt
[...]
550,000 ‘=


```
